classes/PageList.php

```diff
@@ -47,27 +47,22 @@ public function getPageTree($skipCache = false)
     {
         $pages = $this->listPages($skipCache);
         $config = $this->getPagesConfig();
+        
+        // Make the $pages collection an associative array for performance
+        $pagesArray = $pages->keyBy(function ($page) {
+            return $page->getBaseFileName();
+        })->all();
 
-        $iterator = function($configPages) use (&$iterator, &$pages) {
+        $iterator = function($configPages) use (&$iterator, &$pagesArray) {
```

Comment:
@LukeTowers does `$pagesArray` need to be a reference? I don't see anywhere where it's being directly modified.
```suggestion
        $iterator = function($configPages) use (&$iterator, $pagesArray) {
```

Reply:
Yeah, I wasn't sure about myself, just left it a reference since that's what the original code was, but really that shouldn't have been a reference either since it was previously an object (a Collection instance).